cadence/contracts/AxelarGateway.cdc

```diff
@@ -0,0 +1,312 @@
+
+    var allowOperatorshipTransfer = AxelarAuthWeighted.validateProof(message: encodedMessage, operators: operators, weights: weights, threshold: threshold, signatures: signatures)
+
+    let commandsLength = commandIds.length
+      
+    if (commandsLength != commands.length || commandsLength != params.length) {
+      panic("Invalid Commands")
+    }
+
+    var i = 0
+
+    while i < commandsLength {
+      var commandId = commandIds[i]
+      if (self.executedCommandIds.contains(commandId)) {
+        continue
+      }
+
+      let commandHash = Crypto.hash(commands[i].utf8, algorithm: HashAlgorithm.KECCAK_256)
+      let paramHandler = ParamHandler(params: params[i])
+
+      // Transfer OperatorShip is the only command that is directly processed by this contract - unlike other commannds
+      // it is used for managing gateway operators for the Axelar CGP protocol
+      if (commandHash == self.SELECTOR_TRANSFER_OPERATORSHIP && params[i].length == 3) {
+
+        if (allowOperatorshipTransfer) {
+          var transferOperatorShipParams = paramHandler.generateTransferOperatorshipParams()
+
+          if transferOperatorShipParams != nil {
+
+            // we only need to track this was executed, since there's no other processing for this command
+            self.executedCommandIds.append(commandId)
+            AxelarAuthWeighted.transferOperatorship(params: transferOperatorShipParams!)
+            emit Executed(commandId: commandId)
+          }
+        }
+      } else if (commandHash == self.SELECTOR_APPROVE_CONTRACT_CALL && params[i].length == 6) {
+        let approveContractCallParams = paramHandler.generateApproveContractCallParams()
+
+        if approveContractCallParams != nil {
+          // store the CGPCommand Resource for later use
+          let oldCgpCommand <- self.approvedCommands[commandId] <- create CGPCommand(commandId: commandId)
+
+          emit ContractCallApproved (
+            commandId: commandId,
+            sourceChain: approveContractCallParams!.sourceChain,
+            sourceAddress: approveContractCallParams!.sourceAddress,
+            contractAddress: approveContractCallParams!.contractAddress,
+            payloadHash: approveContractCallParams!.payloadHash,
+            sourceTxHash: approveContractCallParams!.sourceTxHash,
+            sourceEventIndex: approveContractCallParams!.sourceEventIndex
+          )
+
+          destroy oldCgpCommand
+        }
+      } else {
+        continue
+      }
+
+      i = i + 1
+    }
+  }
+
+  access(all) fun executeApp(commandId: String, sourceChain: String, sourceAddress: String, contractAddress: String, payload: [UInt8]): Bool {
+    // Check to see if command id has already been executed
+    if (self.executedCommandIds.contains(commandId)) {
+      return false
+    }
+
+    // Validate that contract address string is a valid address
+    let address = Address.fromString(contractAddress) ?? panic("Invalid address")
```

Comment:
There are no guarantees that the address is valid for the network, only that the string is of valid address format. Check out [this util method](https://github.com/green-goo-dao/flow-utils/blob/main/cadence/contracts/AddressUtils.cdc#L55) - there are probably a couple methods that would be helpful to chain here.